src/easylink/step.py

```diff
@@ -1394,15 +1447,17 @@ class LeafConfigurationState(ConfigurationState):
     @property
     def is_combined(self) -> bool:
         """Whether or not this ``Step`` is combined with another ``Step``."""
-        return True if COMBINED_IMPLEMENTATION_KEY in self.pipeline_config else False
+        return COMBINED_IMPLEMENTATION_KEY in self.parent_config[self._step.name]
 
     @property
     def implementation_config(self) -> LayeredConfigTree:
         """The ``Step's`` specific ``Implementation`` configuration."""
         return (
-            self.combined_implementations[self.pipeline_config[COMBINED_IMPLEMENTATION_KEY]]
+            self.combined_implementations[self.parent_config[COMBINED_IMPLEMENTATION_KEY]]
             if self.is_combined
-            else self.pipeline_config["implementation"]
+            else self.parent_config.implementation
+            if not self._step.name in self.parent_config
+            else self.parent_config[self._step.name]["implementation"]
```

Comment:
This definitely feels a little weird, and I'm curious when you would use `self.parent_config.implementation` directly -- can we at least add a comment about this case?

Reply:
I think this is handled appropriately now as well?